drivers/overlay/encryption.go

```diff
@@ -143,17 +161,23 @@ func setupEncryption(localIP, advIP, remoteIP net.IP, vni uint32, em *encrMap, k
 		logrus.Warn(err)
 	}
 
+	var eel []*encrElem
+
 	for i, k := range keys {
-		spis := &spi{buildSPI(advIP, remoteIP, k.tag), buildSPI(remoteIP, advIP, k.tag)}
+		ee := &encrElem{
```

Comment:
Yes, that is an option, given spi and key go hand in hand. I did not go for it because I wanted to contain the refactoring.
I can make the change, it is actually kind of verbose now.